### PR TITLE
Person ordering flexibility

### LINKED ISSUE #1795: More flexible sort order for people
- [ ] Sort order should be a number field rather than a dropdown
- [ ] The person program relationship should also have a sort order, and the one that's not about a person program relationship should not apply in a program situation
- [ ] Also add help text clarifying some of this

### PR DESCRIPTION
**Pull request by chigby**
Fixes #1795

1. There are a number of migrations here, but they're necessary for preserving the existing `sort_priority` values by copying them into the new `sort_order` field (which is now an integer field rather than a list of preset choices).
2. The new field on program/subprogram relationships is also called `sort_order` and it's working pretty well for me, locally. For example, on some subprograms I picked at random:

image

image

### CODE REVIEW

File: newamericadotorg/api/author/tests.py [hunk 1]
```diff
@@ -103,3 +106,152 @@ def test_get_fellows(self):
...
+    def setUpTestData(cls):
+        home_page = PostFactory.create_homepage()
+
+        program1 = PostFactory.create_program(home_page=home_page)
+        program2 = PostFactory.create_program(home_page=home_page)
+        subprogram1 = PostFactory.create_subprogram(program=program1)
+
+        person1 = PostFactory.create_person(
+            {
+                "sort_order": 2,
```
Comment (nmorduch) [inline newamericadotorg/api/author/tests.py:122]: I think you also need to test with people not having a sort order! Same for the person program relationship

**Review by nmorduch — CHANGES_REQUESTED** [review]
See comment re testing!

File: newamericadotorg/api/author/tests.py (hunk 1 above)
Comment (nmorduch) [reply newamericadotorg/api/author/tests.py:122]: To be clear, people with no sort order set go last!

**Comment by chigby** [comment]
Added the tests for absent sort order

**Review by nmorduch — APPROVED** [review]

**Pull request merged**